lib/usd/translators/shading/usdUVTextureReader.cpp

```diff
@@ -0,0 +1,309 @@
+    (fileTextureName)
+    (outAlpha)
+    (outColor)
+    (outColorR)
+    (outColorG)
+    (outColorB)
+    (place2dTexture)
+    (coverage)
+    (translateFrame)
+    (rotateFrame)
+    (mirrorU)
+    (mirrorV)
+    (stagger)
+    (wrapU)
+    (wrapV)
+    (repeatUV)
+    (offset)
+    (rotateUV)
+    (noiseUV)
+    (vertexUvOne)
+    (vertexUvTwo)
+    (vertexUvThree)
+    (vertexCameraOne)
+
+    // UsdUVTexture Input Names
+    (bias)
+    (fallback)
+    (scale)
+    (wrapS)
+    (wrapT)
+
+    // Values for wrapS and wrapT
+    (black)
+    (repeat)
+
+    // UsdUVTexture Output Names
+    ((RGBOutputName, "rgb"))
+    ((RedOutputName, "r"))
+    ((GreenOutputName, "g"))
+    ((BlueOutputName, "b"))
+    ((AlphaOutputName, "a"))
+);
+
+static const TfTokenVector _Place2dTextureConnections = {
+    _tokens->coverage,
+    _tokens->translateFrame,
+    _tokens->rotateFrame,
+    _tokens->mirrorU,
+    _tokens->mirrorV,
+    _tokens->stagger,
+    _tokens->wrapU,
+    _tokens->wrapV,
+    _tokens->repeatUV,
+    _tokens->offset,
+    _tokens->rotateUV,
+    _tokens->noiseUV,
+    _tokens->vertexUvOne,
+    _tokens->vertexUvTwo,
+    _tokens->vertexUvThree,
+    _tokens->vertexCameraOne
+};
+
+PxrMayaUsdUVTexture_Reader::PxrMayaUsdUVTexture_Reader(const UsdMayaPrimReaderArgs& readArgs)
+    : UsdMayaShaderReader(readArgs)
+{
+}
+
+/* virtual */
+bool PxrMayaUsdUVTexture_Reader::Read(UsdMayaPrimReaderContext* context)
+{
+    const auto&    prim = _GetArgs().GetUsdPrim();
+    UsdShadeShader shaderSchema = UsdShadeShader(prim);
+    if (!shaderSchema) {
+        return false;
+    }
+
+    MStatus           status;
+    MFnDependencyNode depFn;
+    if (!(UsdMayaTranslatorUtil::CreateShaderNode(
+              MString(prim.GetName().GetText()),
+              _tokens->file.GetText(),
+              UsdMayaShadingNodeType::Texture,
+              &status,
+              &_mayaObject)
+          && depFn.setObject(_mayaObject))) {
+        // we need to make sure assumes those types are loaded..
+        TF_RUNTIME_ERROR(
+            "Could not create node of type '%s' for shader '%s'.\n",
+            _tokens->file.GetText(),
+            prim.GetName().GetText());
+        return false;
+    }
+
+    // Create place2dTexture:
+    MObject           uvObj;
+    MFnDependencyNode uvDepFn;
+    if (!(UsdMayaTranslatorUtil::CreateShaderNode(
+              _tokens->place2dTexture.GetText(),
+              _tokens->place2dTexture.GetText(),
+              UsdMayaShadingNodeType::Utility,
+              &status,
+              &uvObj)
+          && uvDepFn.setObject(uvObj))) {
+        // we need to make sure assumes those types are loaded..
+        TF_RUNTIME_ERROR(
+            "Could not create node of type '%s' for shader '%s'.\n",
+            _tokens->place2dTexture.GetText(),
+            prim.GetName().GetText());
+        return false;
+    }
+
+    // Connect manually (fileTexturePlacementConnect is not available in batch):
```

Comment:
Code from MEL script fileTexturePlacementConnect copied here since the script is not sourced in batch.